public/js/gulliver.es6.js

```diff
@@ -55,43 +56,57 @@ class Gulliver {
     this.signIn = new SignIn(window, this.config);
     this.signInButton = new SignInButton(this.signIn, document.querySelector('#auth-button'));
 
+    const currentRoute = this.router.findRoute(window.location.href);
+    if (currentRoute) {
+      currentRoute.onAttached();
+    }
+
     // Setup Analytics
     this.analytics = new Analytics(window, this.config);
     this.analytics.trackPageView(window.location.href);
   }
 
-  _addRoute(regexp, transitionStrategy, shellState) {
-    const route = new Route(regexp, transitionStrategy);
+  _addRoute(regexp, transitionStrategy, onRouteAttached, shellState) {
+    const route = new Route(regexp, transitionStrategy, onRouteAttached);
     this.shell.setStateForRoute(route, shellState);
     this.router.addRoute(route);
   }
 
   _setupRoutes() {
     const fadeInOutTransitionStrategy = new FadeInOutTransitionStrategy();
     // Route for `/pwas/add`.
-    this._addRoute(/\/pwas\/add/, fadeInOutTransitionStrategy, {
+    const setupPwaForm = () => {
+      const pwaForm = new PwaForm(window, this.signIn);
+      pwaForm.setup();
+    };
+
+    this._addRoute(/\/pwas\/add/, fadeInOutTransitionStrategy, setupPwaForm, {
       showTabs: false,
       backlink: true,
       subtitle: true
     });
 
+    const setupLighthouseChart = () => {
+      new LighthouseChart().load();
+    };
+
     // Route for `/pwas/[id]`.
-    this._addRoute(/\/pwas\/(\d+)/, fadeInOutTransitionStrategy, {
+    this._addRoute(/\/pwas\/(\d+)/, fadeInOutTransitionStrategy, setupLighthouseChart, {
       showTabs: false,
       backlink: true,
       subtitle: false
     });
 
     // Route for `/?sort=score`.
-    this._addRoute(/\/\?.*sort=score/, fadeInOutTransitionStrategy, {
+    this._addRoute(/\/\?.*sort=score/, fadeInOutTransitionStrategy, null, {
```

Comment:
suggestion, create a dummy function called `nop` and pass it instead of null. This makes it more obvious that there is nothing to do.